packages/samples/headless-react/src/pages/SearchPage.tsx

```diff
@@ -122,7 +129,9 @@ const initialCriterion = criteria[0][1];
 const resultsPerPageOptions = [10, 25, 50, 100];
 
 export class SearchPage extends Component {
-  private readonly engine: Engine;
+  static contextType = AppContext;
```

Comment:
Why does `SearchPage` need a context?

Reply:
Yep, when rendering client-side, no context needs to be passed, but when rendering on the server, it's a way to hold a reference to the engine.